lib/memoist.rb

```diff
@@ -38,14 +38,16 @@ def self.unmemoized_prefix(identifier = nil)
   end
 
   def self.escape_punctuation(string)
-    string = string.is_a?(String) ? string.dup : string.to_s
+    string = string.to_s
 
-    return string unless string.end_with?('?'.freeze, '!'.freeze)
+    return string unless string.end_with?('?', '!')
+
+    string = string.dup if string.frozen?
 
     # A String can't end in both ? and !
-    if string.sub!(/\?\Z/, '_query'.freeze)
```

Comment:
I wonder if it is worth retaining these explicit `.freeze` calls for older versions of ruby (2.2 and below). If not then it would be good to add something to the commit message which explains the reasoning behind dropping them.

Reply:
No you are right, I'll add them back.